flows/waits/time.go

```diff
@@ -0,0 +1,25 @@
+package waits
+
+import (
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+)
+
+const TypeTime string = "time"
+
+type TimeWait struct {
+	Timeout int `json:"timeout" validation:"required,gte=0"`
+}
+
+func (w *TimeWait) Type() string { return TypeMsg }
+
+func (w *TimeWait) Apply(run flows.FlowRun, step flows.Step) {
+	run.ApplyEvent(step, nil, &events.TimeWaitEvent{Timeout: w.Timeout})
+}
+
+// CanResume returns true for a message wait if a message has now been received on this step
+func (w *TimeWait) CanResume(run flows.FlowRun, step flows.Step) bool {
+	return true
```

Comment:
Ok there's two different questions here:

1. Is it valid to have a wait which never waits. To me this is like a `TrueTest` which always returns true or `FalseTest` that always returns false. This is just a special case of a `Wait` which always returns true. We could call it `NeverWait` or `NullWait` or whatever - it's useful for the engine to be able to return to the caller but not actually be waiting for anything. Not sure why that's so weird.
2. Should the engine check the caller waits for timeouts. Maybe. We would need a created_on time on the wait or a timestamp. It does make me a little nervous that either of those routes run into problems if the caller and engine system time are even slightly off.